libs/product/state/src/selectors/composite-product/composite-product.selectors.ts

```diff
@@ -99,6 +103,27 @@ const createCompositeProductSelectors = (): DaffCompositeProductMemoizedSelector
     }),
   );
 
+  const selectCompositeProductDiscountAmount = createSelector(
+    selectProductEntities,
+    selectCompositeProductAppliedOptionsEntities,
+    (products, appliedOptionsEntities, props) => {
+      const product = selectProduct.projector(products, { id: props.id });
+      if(product.type !== DaffProductTypeEnum.Composite) {
+        return undefined;
+      }
+
+      const appliedConfigurationItem = appliedOptionsEntities[product.id].items;
+      if((<DaffCompositeProduct>product).items.filter(item => item.required && appliedConfigurationItem[item.id].value === null).length > 0) {
+        return undefined;
+      }
+      const appliedOptions = getAppliedOptionsForConfiguration(<DaffCompositeProduct>product, appliedConfigurationItem);
+
+      return (<DaffCompositeProduct>product).items.reduce((acc, item) =>
+        daffAdd(acc, daffMultiply(appliedOptions[item.id].discount.amount, appliedOptions[item.id].quantity)),
```

Comment:
Is `appliedOptions[item.id]` guaranteed to exist?

Reply:
nope. fixed and added a test for it.

Reply:
you need to guard the quantity expression as well, right?

I'm not sure optional chaining is the way to go here, I might just use a ternary like:

```ts
appliedOptions[item.id] ? daffAdd(acc, daffMultiply(appliedOptions[item.id].discount.amount, appliedOptions[item.id].quantity)) : acc
```

so that it avoids calling arithmetic function unless its actually necessary.

Reply:
Turned out that `appliedOptions[item.id]` is guaranteed to exist, but your way makes more sense. I just needed to guard up to `appliedOptions[item.id].discount.amount`